### PR TITLE
STOR-2017: Bump all deps for 4.18

### PR DESCRIPTION
**Pull request by rhrmo**
Bump all deps for 4.18

Notes:
* Updated calls for functions coming from library-go
* Added spec.unhealthyEvictionPolicy: AlwaysAllow to all pdbs

### CODE REVIEW

File: pkg/operator/starter.go [hunk 1]
```diff
@@ -127,39 +127,17 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
...
-
 	gvr := operatorv1.SchemeGroupVersion.WithResource("csisnapshotcontrollers")
 	gvk := operatorv1.SchemeGroupVersion.WithKind("CSISnapshotControllers")
-	guestOperatorClient, dynamicInformers, err := goc.NewClusterScopedOperatorClientWithConfigName(clock.RealClock{}, guestKubeConfig, gvr, gvk, "cluster", extractApplySpec, extractApplyStatus)
+	guestOperatorClient, dynamicInformers, err := goc.NewClusterScopedOperatorClientWithConfigName(
+		clock.RealClock{},
+		guestKubeConfig,
+		gvr,
+		gvk,
+		string(opv1.GCPPDCSIDriver),
```
Comment (jsafrane) [inline pkg/operator/starter.go]: this should be "cluster", not "pd.csi.storage.gke.io".

Comment (rhrmo) [reply pkg/operator/starter.go]: thanks a lot!

File: pkg/operator/starter.go [hunk 2]
```diff
@@ -745,3 +757,33 @@ func withVolumeGroupSnapshotWebhook(enabled bool) dc.DeploymentHookFunc {
 		return nil
 	}
 }
+
+func extractOperatorSpec(obj *unstructured.Unstructured, fieldManager string) (*applyoperatorv1.OperatorSpecApplyConfiguration, error) {
+	castObj := &operatorv1.ClusterCSIDriver{}
```
[the remark this answers is not in the document]
Comment (rhrmo) [reply pkg/operator/starter.go]: Thank you for noticing, yes it should AFAIK

File: pkg/operator/starter.go [hunk 3]
```diff
@@ -745,3 +757,33 @@ func withVolumeGroupSnapshotWebhook(enabled bool) dc.DeploymentHookFunc {
...
+		return nil, nil
+	}
+	return &ret.Spec.OperatorSpecApplyConfiguration, nil
+}
+func extractOperatorStatus(obj *unstructured.Unstructured, fieldManager string) (*applyoperatorv1.OperatorStatusApplyConfiguration, error) {
+	castObj := &operatorv1.CSISnapshotController{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, castObj); err != nil {
+		return nil, fmt.Errorf("unable to convert to CSISnapshotController: %w", err)
+	}
+	ret, err := applyoperatorv1.ExtractCSISnapshotController(castObj, fieldManager)
```
Comment (mpatlasov) [inline pkg/operator/starter.go:780]: s/ExtractCSISnapshotController/ExtractCSISnapshotControllerStatus

Comment (rhrmo) [reply pkg/operator/starter.go:780]: Thank you

**Comment by rhrmo** [comment]
/payload-job periodic-ci-openshift-release-master-ci-4.18-e2e-aws-ovn-upgrade

**Comment by rhrmo** [comment]
/payload-job periodic-ci-openshift-release-master-ci-4.18-upgrade-from-stable-4.17-e2e-gcp-ovn-rt-upgrade

**Pull request merged**